web/src/components/molecules/Schema/FieldModal/FieldUpdateModal/index.tsx

```diff
@@ -0,0 +1,200 @@
+import styled from "@emotion/styled";
+import React, { useCallback, useEffect } from "react";
+
+import Checkbox from "@reearth-cms/components/atoms/Checkbox";
+import Form from "@reearth-cms/components/atoms/Form";
+import Icon from "@reearth-cms/components/atoms/Icon";
+import Input from "@reearth-cms/components/atoms/Input";
+import Modal from "@reearth-cms/components/atoms/Modal";
+import Tabs from "@reearth-cms/components/atoms/Tabs";
+import TextArea from "@reearth-cms/components/atoms/TextArea";
+import { SchemaFieldTypePropertyInput } from "@reearth-cms/gql/graphql-client-api";
+
+import { fieldTypes } from "../../fieldTypes";
+import { Field, FieldType } from "../../types";
+import FieldDefaultProps from "../FieldDefaultProps";
+import FieldValidationProps from "../FieldValidationProps";
+
+export interface FormValues {
+  fieldId: string;
+  title: string;
+  description: string;
+  key: string;
+  typeProperty: SchemaFieldTypePropertyInput;
+}
+
+export interface Props {
+  open?: boolean;
+  selectedType: FieldType | null;
+  selectedField?: Field | null;
+  onClose?: (refetch?: boolean) => void;
+  onSubmit?: (values: FormValues) => Promise<void> | void;
+}
+
+const initialValues: FormValues = {
+  fieldId: "",
+  title: "",
+  description: "",
+  key: "",
+  typeProperty: { text: { defaultValue: "", maxLength: 0 } },
+};
+
+const FieldUpdateModal: React.FC<Props> = ({
+  open,
+  onClose,
+  onSubmit,
+  selectedType,
+  selectedField,
+}) => {
+  const [form] = Form.useForm();
+  const { TabPane } = Tabs;
+
+  useEffect(() => {
+    form.setFieldsValue({
+      fieldId: selectedField?.id,
+      title: selectedField?.title,
+      description: selectedField?.description,
+      key: selectedField?.key,
+      defaultValue:
+        selectedField?.typeProperty.defaultValue ||
+        selectedField?.typeProperty.selectDefaultValue ||
+        selectedField?.typeProperty.integerDefaultValue,
+      min: selectedField?.typeProperty.min,
+      max: selectedField?.typeProperty.max,
+      maxLength: selectedField?.typeProperty.maxLength,
+    });
+  }, [form, selectedField]);
+
+  const handleSubmit = useCallback(() => {
+    form
+      .validateFields()
+      .then(async values => {
+        if (selectedType === "Text") {
+          values.typeProperty = {
+            text: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "TextArea") {
+          values.typeProperty = {
+            textArea: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "MarkdownText") {
+          values.typeProperty = {
+            markdownText: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
```

Comment:
Again, if if if if if isn't really nice for the developer to read and we can't be sure there won't be unforseen results compared with an else/if or switch, which will finish once something is true.

Reply:
@KaWaite , I've changed it to `is/else`